sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -238,6 +304,69 @@ declare global {
   type F32 = F<32>;
   type F64 = F<64>;
 
+  function INT<const N extends string>(name: N): INT<N>;
+  function INT<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals: { allowable_ranges?: VariableRange[]; allowable_values?: unknown[]; sc_name?: string },
+  ): INT<N>;
+  function INT<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals?: { allowable_ranges?: VariableRange[]; allowable_values?: unknown[]; sc_name?: string },
+  ): INT<N>;
+
+  function UINT<const N extends string>(name: N): UINT<N>;
+  function UINT<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals: { allowable_ranges?: VariableRange[]; allowable_values?: unknown[]; sc_name?: string },
+  ): UINT<N>;
+  function UINT<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals?: { allowable_ranges?: VariableRange[]; allowable_values?: unknown[]; sc_name?: string },
+  ): UINT<N>;
+
+  function FLOAT<const N extends string>(name: N): FLOAT<N>;
+  function FLOAT<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals: { allowable_ranges?: VariableRange[]; allowable_values?: unknown[]; sc_name?: string },
+  ): FLOAT<N>;
+  function FLOAT<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals?: { allowable_ranges?: VariableRange[]; allowable_values?: unknown[]; sc_name?: string },
+  ): FLOAT<N>;
+
+  // @ts-ignore : 'VariableRange' found in JSON Spec
+  function STRING<const N extends string>(name: N): STRING<N>;
+  // @ts-ignore : 'VariableRange' found in JSON Spec
+  function STRING<const N extends string>(
+      name: N,
+      optionals: { allowable_values?: unknown[]; sc_name?: string },
+  ): STRING<N>;
+  function STRING<const N extends string>(
+      name: N,
+      // @ts-ignore : 'VariableRange' found in JSON Spec
+      optionals?: { allowable_values?: unknown[]; sc_name?: string },
+  ): STRING<N>;
+
+  function ENUM<const N extends string, const E extends string>(name: N, enum_name: E): ENUM<N, E>;
```

Comment:
Long term, we can probably type check the enum variant against the enum name, but that would be backwards compatible with _valid_ sequences.

Reply:
Do you mean something like this?

`ENUM('direction', ['fromStem', 'fromPeel'])`

`C.PEEL_BANANA({peelDirection : locals.direction})`

Reply:
I think I misunderstood here. Is this is defining a _new_ enum, not declaring a use of an enum from the command dictionary?